tests/db/test_controls_workflows_mapping.py

```diff
@@ -0,0 +1,176 @@
+        assert mapping.workflow_id == workflow.workflow_id
+
+        # Refresh the relationships
+        await session.refresh(control, ["control_workflow_mappings"])
+        await session.refresh(workflow, ["control_workflow_mappings"])
+
+        # Get the related objects through the mapping
+        control_mappings = control.control_workflow_mappings
+        workflow_mappings = workflow.control_workflow_mappings
+
+        assert len(control_mappings) == 1
+        assert len(workflow_mappings) == 1
+        assert control_mappings[0].workflow_id == workflow.workflow_id
+        assert workflow_mappings[0].control_id == control.control_id
+
+
+@pytest.mark.asyncio
+async def test_cascade_delete_workflow(store: AdmyralStore):
+    async with store.async_session_maker() as session:
+        await store.clean_up_workflow_data_of(TEST_USER_ID)
+        await store.clean_up_controls_data(TEST_USER_ID)
+        _, workflow, _ = await setup_scenario(session)
+
+        # Create a control result
+        control_result = WorkflowControlResultsSchema(
+            workflow_id=workflow.workflow_id,
+            run_id="run123",
+            result=True,
+            user_id=TEST_USER_ID,
+        )
+        session.add(control_result)
+        await session.commit()
+
+        # Verify everything is created
+        mapping_result = await session.exec(select(ControlsWorkflowsMappingSchema))
+        mappings = mapping_result.all()
+        assert len(mappings) == 1
+
+        results_result = await session.exec(select(WorkflowControlResultsSchema))
+        results = results_result.all()
+        assert len(results) == 1
+
+        # First delete the control results
+        results_to_delete = await session.exec(
+            select(WorkflowControlResultsSchema).where(
+                WorkflowControlResultsSchema.workflow_id == workflow.workflow_id
+            )
+        )
+        for result in results_to_delete.all():
+            await session.delete(result)
+
+        # Then delete the workflow mappings
+        mappings_to_delete = await session.exec(
+            select(ControlsWorkflowsMappingSchema).where(
+                ControlsWorkflowsMappingSchema.workflow_id == workflow.workflow_id
+            )
+        )
+        for mapping in mappings_to_delete.all():
+            await session.delete(mapping)
+
+        # Finally delete the workflow
+        await session.delete(workflow)
+        await session.commit()
+
+        # Verify cascading deletes
+        mapping_result = await session.exec(select(ControlsWorkflowsMappingSchema))
+        mappings = mapping_result.all()
+        assert len(mappings) == 0
+
+        results_result = await session.exec(select(WorkflowControlResultsSchema))
+        results = results_result.all()
+        assert len(results) == 0
+
+        control_result = await session.exec(select(ControlSchema))
+        controls = control_result.all()
+        assert len(controls) == 1
```

Comment:
Can we also test the other direction? Delete a control, then it should remove the corresponding rows from the mapping table